api/utils.ts

```diff
@@ -0,0 +1,21 @@
+export function allDoNotContainDangerousCharacters(values: string[]) {
+  for (const value of values) {
+    if (!hasNoDangerousCharacters(value)) {
+      // if one value contains dangerous characters, return false
+      return false;
+    }
+  }
+  return true;
+}
+
+export function hasNoDangerousCharacters(value: string): boolean {
+  // the below only checks for the characters in the string
+  // return /^[^%<>^$//]+$/.test(value);
+  // substring approach below preferred
+  return /^(?!.*\/\*|.*\/\/|.*;|.*&|.*\?|.*<|.*\^|.*>).+$/.test(value);
```

Comment:
I don't know why this is so complex. What is wrong with /[^%<>^$//]/.test(value)
This function appears to return true if any of the chars are present, so has the wrong logical name

Reply:
This regEx allows for removing dangerous substrings too e.g. '.*\/\/' removes '//' as being possible in any of the data and '.*\/\*' removes '/*'. The regEx you mention can only be used to remove dangerous characters, not substrings. In the list of dangerous possibilities you gave me some of them were '//' for example and removing '/' is not necessary. I can change it to the regEx above but we would not have that flexibility